### PR TITLE
Echo server

### PR DESCRIPTION
**Pull request by Kotauror**
Hi! 

This is a PR for the echo server - no client in this PR.

### CODE REVIEW

**Review by sarahabimay — COMMENTED** [review]
Looks good.  I can't remember if you need to commit the *.iml file do  you?  Maybe you do to run in IntelliJ, but not for anywhere else..  Maybe that's where Gradle would come in.
Some of your commit messages are a bit long and can't be read without opening the commit first.  Try to avoid that.  If you have more to say then add it in the 'descriptions' section.

File: src/com/company/EchoServer.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
+package com.company;
+
+import java.io.IOException;
+import java.net.ServerSocket;
+import java.net.Socket;
+
+public class EchoServer {
+
+    private ServerSocket serverSocket;
+    private int portNumber;
```
Comment (sarahabimay) [inline src/com/company/EchoServer.java:10]: Move this above ServerSocket to alphabetise

File: src/com/company/EchoServer.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    private ServerMessenger serverMessenger;
+
+    public EchoServer(ServerSocket serverSocket, int portNumber, ServerMessenger serverMessenger) {
```
Comment (sarahabimay) [inline src/com/company/EchoServer.java:13]: portNumber is an attribute of ServerSocket and therefore could be accessed via a getter in ServerSocket whenever it is needed in EchoServer e.g. to print it out

File: src/com/company/EchoServer.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+    public boolean runServer() {
+        return true;
+    }
+
+    private void connectWithSocket() {
+        try {
+            Socket clientSocket = accept();
+            serverMessenger.informOfNewSocket();
+            echo(clientSocket);
```
Comment (sarahabimay) [inline src/com/company/EchoServer.java:34]: clientSocket/accept() could be inlined here

File: src/com/company/EchoServer.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+        } catch (IOException exception) {
+            serverMessenger.informOfException(this.portNumber, exception.getMessage());
+        }
+    }
+
+    private Socket accept() throws IOException {
```
Comment (sarahabimay) [inline src/com/company/EchoServer.java:40]: You could get rid of this method seeing as it only contains one line and that line is already as informative as the method name `accept()` anyways.  Unless this method name gives any more information to the reader or has a few lines of related code, then it is unnecessary.

File: src/com/company/EchoServer.java [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+        return this.serverSocket.accept();
+    }
+
+    private static void echo(Socket clientSocket) throws IOException {
+        IOHandler iOHandler = new IOHandler(clientSocket);
+        String inputLine;
+        while ((inputLine = iOHandler.readFromSocket()) != null) {
+                iOHandler.printToSocket(inputLine);
```
Comment (sarahabimay) [inline src/com/company/EchoServer.java:48]: Indentation is off

Comment (Kotauror) [reply src/com/company/EchoServer.java:48]: done.

File: src/com/company/EchoServer.java (hunk 4 above)
Comment (Kotauror) [reply src/com/company/EchoServer.java:40]: done.

File: src/com/company/EchoServer.java (hunk 3 above)
Comment (Kotauror) [reply src/com/company/EchoServer.java:34]: done.

File: src/com/company/EchoServer.java (hunk 2 above)
Comment (Kotauror) [reply src/com/company/EchoServer.java:13]: done.

**Pull request merged**